actnow/petitions/models.py

```diff
@@ -0,0 +1,71 @@
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+
+from actnow.accounts.models import ActNowUser
+from actnow.db.models import TimestampedModelMixin
+
+
+class Petition(TimestampedModelMixin):
+    title = models.CharField(
+        _("title"),
+        max_length=255,
+    )
+    description = models.CharField(
+        _("details of petition"),
+        max_length=1024,
+        blank=True,
+    )
+    owner = models.ForeignKey(
+        ActNowUser,
+        on_delete=models.SET_NULL,
+        null=True,
+    )
+    recipients = models.CharField(
+        _("recipients"),
+        max_length=255,
+    )
+    problem_statement = models.CharField(
+        _("problem statement"),
+        max_length=1024,
+    )
+    number_of_signatures_required = models.PositiveIntegerField(
+        _("number of signatures required"),
+        default=0,
+    )
+    image = models.ImageField(
+        _("image"),
+        blank=True,
+    )
+    video = models.FileField(
+        _("video"),
+        blank=True,
+    )
+
+    def __str__(self):
+        return self.title
+
+
+class PetitionSignature(TimestampedModelMixin):
+    petition = models.ForeignKey(
+        Petition,
+        on_delete=models.CASCADE,
+    )
+    signatory = models.ForeignKey(
+        ActNowUser,
+        on_delete=models.SET_NULL,
```

Comment:
(Just a thought) What will happen to these signatures? Will they be counted as valid once an account is deleted? If so, won't one person be able to signup, sign the petition, and then delete their account(I know accounts won't be fully deleted but will just be marked as inactive) and then do this all over again?

Reply:
@esirK changed this to `CASCADE`